synapse/config/emailconfig.py

```diff
@@ -74,14 +74,75 @@ def read_config(self, config):
             "account_validity", {},
         ).get("renew_at")
 
-        if self.email_enable_notifs or account_validity_renewal_enabled:
+        self.email_enable_password_reset_from_is = email_config.get(
+            "enable_password_reset_from_is", False,
+        )
+        self.enable_password_resets = (
+            self.email_enable_password_reset_from_is
+            or (not self.email_enable_password_reset_from_is and email_config != {})
+        )
+        if email_config == {} and not self.email_enable_password_reset_from_is:
+            logger.warn(
+                "User password resets have been disabled due to lack of email config."
+            )
+
+        self.email_validation_token_lifetime = email_config.get(
+            "validation_token_lifetime", 15 * 60,
```

Comment:
Use `self.parse_duration` so that you can say `15m` in the config. We probably want to keep this valid for, like, `1h` at least, since email can be quite slow at times.

Reply:
Also, should be in milliseconds for consistency